cou/utils/app_utils.py

```diff
@@ -126,3 +109,101 @@ def validate_ovn_support(version: str) -> None:
                 "ovn-upgrade-2203.html"
             )
         )
+
+
+# Private functions
+async def _get_required_osd_release(unit: str, model: COUModel) -> str:
```

Comment:
Both `_get_required_osd_release` and ` _get_current_osd_release` they share the same structure:

1. run a command
2. check if the command returned 0
3. raise RunUpgradeError if not 0
4. raise  RunUpgradeError if JujuError happens.

Can't we share a single generic function that does this job?

Reply:
They do have very similar structures, and I would love to merge them. However, their error messages are tailored, so it is hard to do so. Do you have any suggestions?

Reply:
taking a closer look, set_require_osd_release_option also has a similar structure. I would try something like:

```python
async def _run_ceph_mon_command(command, unit, model: COUModel, run_upgrade_err_msg):
    logger.debug("Running '%s' on '%s'", command, unit)
    try:
        result = await model.run_on_unit(unit_name=unit, command=command, timeout=600)
        if str(result["Code"]) != "0":
            raise RunUpgradeError(run_upgrade_err_msg) from CommandRunFailed(cmd=command, result=result)
        return result["Stdout"]

    except JujuError as exc:
        raise RunUpgradeError(run_upgrade_err_msg) from exc

async def _get_required_osd_release(unit: str, model: COUModel) -> str:
    cmd = "ceph versions -f json"
    err_msg = f"Cannot determine the current value of require_osd_release on ceph-mon unit '{unit}'."
    return json.loads(wait _run_ceph_mon_command(cmd, unit, err_msg)).get("require_osd_release", "")
```

This can increase substantially the readability and apply DRY

Reply:
Updated